.github/workflows/release.yml

```diff
@@ -0,0 +1,127 @@
+name: release
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - 'v*'
+
+permissions: {}
+
+jobs:
+  build:
+    if: (github.repository == 'rubocop/rubocop')
+    permissions:
+      contents: read
+    runs-on: ubuntu-latest
+    outputs:
+      artifact-id: ${{ steps.upload-artifact.outputs.artifact-id }}
+      gem_name: ${{ steps.build.outputs.gem_name }}
+      gem_version: ${{ steps.build.outputs.gem_version }}
+      gem_path: ${{ steps.build.outputs.gem_path }}
+
+    steps:
+      - name: checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - uses: ruby/setup-ruby@d8d83c3960843afb664e821fed6be52f37da5267 # pin@v1.231.0
```

Comment:
Usually the workflows here don't pin such specific versions but it makes sense for this one I guess.

Dependabot does updates here (https://github.com/rubocop/rubocop/blob/master/.github/dependabot.yml), how would it handle these, would it make PRs for every minor/patch version?

Reply:
Yep that is correct, dependabot will still open PRs to bump for new versions as usual. You would absolutely want to pin to an exact commit sha (or immutable action that I will talk about below) for security / stability.